io/include/detray/io/common/detail/detector_components_io.hpp

```diff
@@ -56,15 +56,16 @@ class detector_component_writers final {
     /// writers, while using the name map @param names for the detector
     virtual void write(const detector_t& det,
                        const typename detector_t::name_map& names,
+                       const std::filesystem::path& file_path,
```

Comment:
You can put this as the last argument with default path `"./"`.

Reply:
Done